tests/core/pyspec/eth2spec/test/phase0/unittests/fork_choice/test_on_block.py

```diff
@@ -274,14 +279,19 @@ def test_on_block_outside_safe_slots_but_finality(spec, state):
     next_epoch(spec, state)
     spec.on_tick(store, store.time + state.slot * spec.SECONDS_PER_SLOT)
     state, store, last_signed_block = apply_next_epoch_with_attestations(spec, state, store)
+    last_block_root = hash_tree_root(last_signed_block.message)
+
+    # Mock fictitious justified checkpoint in store
+    store.justified_checkpoint = spec.Checkpoint(
+        epoch=spec.compute_epoch_at_slot(last_signed_block.message.slot),
+        root=spec.Root("0x4a55535449464945440000000000000000000000000000000000000000000000")
```

Comment:
why not use htr of `last_block_root` for `justified_checkpoint.root`?

Reply:
We want to trigger this piece of code from `should_update_justified_checkpoint`:
```python
if not get_ancestor(store, new_justified_checkpoint.root, justified_slot) == store.justified_checkpoint.root:
        return False
```

If we use `last_block_root` & `spec.compute_epoch_at_slot(last_signed_block.message.slot)` to construct `store.justified_checkpoint`, then the newly constructed blocks will not trigger this, and `should_update_justified_checkpoint` returns `True` because the new justified checkpoint is a descendant of the current one in the store.

Reply:
Filling in a fictitious root ensures that this `get_ancestor` check fails and `should_update_justified_checkpoint` returns `False`, creating the test scenario we want